src/js/api/FirebaseApiClient.js

```
@@ -0,0 +1,491 @@
+            this._log(`Received signal for ${userId} from ${data.from}`);
+            
+            // Call the onSignal handler
+            this.onSignal?.(data.from, data.signal);
+            
+            // Mark as processed
+            await updateDoc(change.doc.ref, {
+              processed: true,
+              retrievedAt: serverTimestamp()
+            });
+          }
+        });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)